Providers/CoinsProviderConfig.jsx

```diff
@@ -0,0 +1,123 @@
+// /* eslint-disable react/destructuring-assignment */
+import React, { Component, createContext } from "react";
+import { AsyncStorage, Alert } from "react-native";
+import axios from "react-native-axios";
+
+const { Provider, Consumer } = createContext();
+
+class CoinsProvider extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      coins: null,
+      date: null
+    };
+  }
+
+  async componentDidMount() {
+    const dateResult = await this.loadFromStorage();
+    try {
+      const resp = await axios.get("http://worldtimeapi.org/api/ip");
+
+      const date = new Date(resp.data.datetime);
+      const storedDate = new Date(dateResult);
```

Comment:
```suggestion
      const storedDate = new Date(await this.loadFromStorage());
```
nie musisz tego tak dziwnie nazywać wtedy